File: backend/src/main/java/hu/szakdoga/backend/timetable/data/dto/MainTaskDTO.java
package hu.szakdoga.backend.timetable.data.dto;

import java.util.Date;

public class MainTaskDTO {
    public final Long id;
    public final String name;
    public final boolean fulfilled;
    public final Date deadline;
    public final String note;
    public final String type;
    public final Long lessonId;

    public MainTaskDTO(Long id, String name, boolean fulfilled, Date deadline, String note, String type, Long lessonId) {
        this.id = id;
        this.name = name;
        this.fulfilled = fulfilled;
        this.deadline = deadline;
        this.note = note;
        this.type = type;
        this.lessonId = lessonId;
    }
}

File: frontend/src/app/shared/model/timetable/dto/main-task.dto.ts
export class MainTaskDto {
    public readonly id: number | null;
    public readonly name: string;
    public readonly fulfilled: boolean;
    public readonly deadline: Date;
    public readonly note: string;
    public readonly type: string;
    public readonly lessonId: number;

    constructor(
        name: string,
        fulfilled: boolean,
        deadline: Date,
        note: string,
        type: string,
        lessonId: number,
        id: number | null = null
    ) {
        this.id = id;
        this.name = name;
        this.fulfilled = fulfilled;
        this.deadline = deadline;
        this.note = note;
        this.type = type;
        this.lessonId = lessonId;
    }
}
